src/app/services/teacher-round.service.ts

```diff
@@ -25,9 +23,15 @@ export class TeacherRoundService {
     // Link up observables so that the timer state gets sent to the current
     // round in Firebase. (But don't do anything when the current round is
     // null.)
-    combineLatest([this.teacherSessionService.currentRoundPath$, this.timerService.running$]).pipe(
+    // Also, record pause & play events based on state of running observable from timer service and
+    // the time of its occurrence.
+    combineLatest([this.teacherSessionService.currentRoundPath$, this.timerService.running$, this.timerService.currentTime$]).pipe(
       filter(([roundPath]) => roundPath !== null),
-    ).subscribe(([roundPath, running]) => {
+    ).subscribe(([roundPath, running, time]) => {
+      this.firebaseService.addHostEvent(
+        roundPath,
+        { eventType: ( running ? HostEventType.Play : HostEventType.Pause ), timePeriod: time.time}
+      );
```

Comment:
Uffda, this is recording the round state every tick :grimacing: I'm sorry about that, Utkarsh!

(Maybe if we put `addHostEvent()` in its own subscription, we could it through a `distinctUntilChanged()` first? :man_shrugging:)

Sorry to be the bearer of bad news!

Reply:
(If you want to be extra-especially-cool, you could even capture that behavior in a unit test 😉)

Reply:
The issue here is subscribing to current time in the combine latest. You want to actually only check the current time when the running state has changed.

Reply:
Ah, with a `pipe(take(1))` or something!